### PR TITLE
feat: use href attribute in Algolia to navigate to result screen

### PR DESCRIPTION
**Pull request by dblandin**
The type of this PR is: **Enhancement**

### Description

_Note: This updates a screen currently behind a feature flag_

Use the `href` attribute in Algolia objects to navigate to search results.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- update new improved search screen (behind feature flag) to use `href` to navigate to results - dblandin

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

</details>

## Screenshots

Simulator Screen Shot - iPhone 12 Pro Max - 2021-09-08 at 18 41 38

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (update new improved search screen (behind feature flag) to use `href` to navigate to results - dblandin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1802c54da272a2ae9106aa57aec904df93b33d34
</p>

File: src/lib/Scenes/Search2/Search2.tsx [hunk 1]
```diff
@@ -75,35 +75,36 @@ const Highlight = connectHighlight(({ highlight, attribute, hit, highlightProper
...
-  nationality: string
-  visible_to_public: boolean
-  fair_ids: string[]
-  partner_ids: string[]
+interface AlgoliaSearchResult {
+  href: string
   image_url: string
-  birth_year: string
+  name: string
+  objectID: string
```
Comment (gkartalis) [inline src/lib/Scenes/Search2/Search2.tsx:82]: 🤩

**Review by gkartalis — APPROVED** [review]
Looks great!

**Comment by gkartalis** [comment]
#squashOnGreen

**Pull request merged**